store/src/java/com/zimbra/cs/service/account/ChangePassword.java

```diff
@@ -101,6 +103,21 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
             AuthProvider.validateAuthToken(prov, zsc.getAuthToken(), false);
             prov.setPassword(acct, newPassword, true);
             acct.setVirtualAccountInitialPasswordSet(true);
+        } else if (authTokenEl != null) {
+            try {
+                AuthToken at = AuthProvider.getAuthToken(authTokenEl, acct);
+                Account authTokenAcct = AuthProvider.validateAuthToken(prov, at, false, AuthToken.Usage.RESET_PASSWORD);
+                if (authTokenAcct != null && AuthToken.Usage.RESET_PASSWORD == at.getUsage()) {
+                    prov.changePassword(acct, oldPassword, newPassword, dryRun);
+                    try {
+                        at.deRegister();
+                    } catch (AuthTokenException e) {
+                        throw ServiceException.FAILURE("cannot de-register reset password auth token", e);
+                    }
+                }
```

Comment:
Maybe re-arrange this if-else block to log the auth token issues, missing account or wrong type. Mostly wrong type.